### ISSUE TITLE
EACCES Error with the Addon

### ISSUE THREAD

**Issue by schmierlappe** [matt8707/addon-ha-fusion#5]
Running Ha Supervisor with Debian Bookworm

Error in the log

s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
Starting Fusion...
node:events:492
      throw er; // Unhandled 'error' event
      ^

Error: listen EACCES: permission denied 0.0.0.0:5050
    at Server.setupListenHandle [as _listen2] (node:net:1855:21)
    at listenInCluster (node:net:1920:12)
    at doListen (node:net:2069:7)
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
Emitted 'error' event on Server instance at:
    at emitErrorNT (node:net:1899:8)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  code: 'EACCES',
  errno: -13,
  syscall: 'listen',
  address: '0.0.0.0',
  port: 5050
}

Node.js v20.8.1
s6-rc: info: service legacy-services: stopping
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped

**Comment by jasonmadigan** [comment]
Seeing same, also running HA Supervisor on Debian Bookworm

**Comment by Gerrett84** [comment]
same here. 

Ha-Fusion docker Container is running. (Accessible at hassio_url:5050)

**Comment by matt8707** [comment]
So is this isolated to debian bookworm?

* try another port 5678:5050
* permission denied, try sudo

**Comment by jasonmadigan** [comment]
> So is this isolated to debian bookworm?
> 
> * try another port 5678:5050
> * permission denied, try sudo

Installing via addon, didn't see any options to configure. Nothing else listening on 5050 on the machine.

Re: trying sudo - don't think I can, unless the addon can be configured differently?

**Comment by matt8707** [comment]
Ah, addon.. no other port is interfering and all other addons work fine? So it's "permission denied" somehow. some more logs would be great

**Comment by jasonmadigan** [comment]
All other addons OK yes. Similar logs here to @schmierlappe:

```
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
Starting Fusion...
node:events:492
      throw er; // Unhandled 'error' event
      ^

Error: listen EACCES: permission denied 0.0.0.0:5050
    at Server.setupListenHandle [as _listen2] (node:net:1855:21)
    at listenInCluster (node:net:1920:12)
    at Server.listen (node:net:2008:7)
    at Function.listen (/rootfs/node_modules/express/lib/application.js:635:24)
    at file:///rootfs/server.js:27:5
Emitted 'error' event on Server instance at:
    at emitErrorNT (node:net:1899:8)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  code: 'EACCES',
  errno: -13,
  syscall: 'listen',
  address: '0.0.0.0',
  port: 5050
}

Node.js v20.8.1
s6-rc: info: service legacy-services: stopping
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

**Comment by Gerrett84** [comment]
Same here

```
_s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
Starting Fusion...
node:events:492
      throw er; // Unhandled 'error' event
      ^

Error: listen EACCES: permission denied 0.0.0.0:5050
    at Server.setupListenHandle [as _listen2] (node:net:1855:21)
    at listenInCluster (node:net:1920:12)
    at Server.listen (node:net:2008:7)
    at Function.listen (/rootfs/node_modules/express/lib/application.js:635:24)
    at file:///rootfs/server.js:27:5
Emitted 'error' event on Server instance at:
    at emitErrorNT (node:net:1899:8)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  code: 'EACCES',
  errno: -13,
  syscall: 'listen',
  address: '0.0.0.0',
  port: 5050
}

[Node.js v20.8.1
s6-rc: info: service legacy-services: stopping
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner  successfully stopped]

```

If I don't run Ha-Fusion as addons, it works.

![Screenshot_20231231-000230.png](https://github.com/matt8707/addon-ha-fusion/assets/1213908/d8636f0f-406c-4da9-b7e3-29d5a51680f6)

**Comment by Gerrett84** [comment]
> So is this isolated to debian bookworm?
> 
> * try another port 5678:5050
> * permission denied, try sudo

I changed the port and now it starts Would it be possible for you to make the port configurable?

Thanks :-)

https://github.com/Gerrett84/addon-ha-fusion/commit/0950bb016e97297ba04f3672e66a070783a8fe47

**Comment by Gerrett84** [comment]
I just changed the port in dockersfile in my fork :-)

Here:
https://github.com/Gerrett84/addon-ha-fusion/commit/0950bb016e97297ba04f3672e66a070783a8fe47

**Comment by matt8707** [comment]
@Gerrett84 Yes but you also need to comment out image in config.yaml, otherwise you just pull the image from dockerhub without your changes, thats why I'm asking

**Comment by Gerrett84** [comment]
@matt8707 Why? I only change the container port configuration and nothing in the image.

![Screenshot_20240102-214000.png](https://github.com/matt8707/addon-ha-fusion/assets/1213908/af62d2a0-e783-4726-86ee-4deb146e48d4)

**Comment by matt8707** [comment]
What do you mean? a Dockerfile is for building the image, you just linked to you changing the env in the image, What is that you have a screenshot of?

**Comment by matt8707** [comment]
Please try version 2024.1.0. I've set the port to the default ingress value, hoping that it fixes the issue. https://github.com/matt8707/addon-ha-fusion/commit/bf3e61597b6a7de2d1aa66067b1e9e19a5829841

**Comment by Gerrett84** [comment]
> What do you mean? a Dockerfile is for building the image, you just linked to you changing the env in the image, What is that you have a screenshot of?

Sure, the image will be built with the configuration from the Dockersfile. Now I understand what you were getting at. But I don't have to comment on it.

**Comment by Gerrett84** [comment]
> Please try version 2024.1.0. I've set the port to the default ingress value, hoping that it fixes the issue. https://github.com/matt8707/addon-ha-fusion/commit/bf3e61597b6a7de2d1aa66067b1e9e19a5829841 do a complete uninstall and install

Unfortunately, port 8099 is already occupied by Nextcloud ;-)
But even after quitting Nextcloud I get the same error again

**Comment by Gerrett84** [comment]
> Please try version 2024.1.0. I've set the port to the default ingress value, hoping that it fixes the issue. https://github.com/matt8707/addon-ha-fusion/commit/bf3e61597b6a7de2d1aa66067b1e9e19a5829841 do a complete uninstall and install

Unfortunately, port 8099 is already occupied by Nextcloud ;-)
But even after quitting Nextcloud I get the same error again

**Comment by matt8707** [comment]
Thanks, so changing port has no effect on this issue. Can you make a detailed explanation on how you made it work then?

**Comment by churchofnoise** [comment]
Confirmation here that the error persists after the update, only change in the log is the port

**Comment by Gerrett84** [comment]
> Thanks, so changing port has no effect on this issue. Can you make a detailed explanation on how you made it work then?

I no longer believe that the port itself is the problem. The fact that it now works for me is not because I changed the port.
If I do the following the addon starts without me having to change the port or anything else: 
Add your repo and install the addon - Delete the hassio/addon/git folder from the addon, uninstall the addon and reinstall it directly. And the addon starts. Do not know why.

**Comment by Gerrett84** [comment]
> Thanks, so changing port has no effect on this issue. Can you make a detailed explanation on how you made it work then?

I no longer believe that the port itself is the problem. The fact that it now works for me is not because I changed the port.
If I do the following the addon starts without me having to change the port or anything else: 
Add your repo and install the addon - Delete the hassio/addon/git folder from the addon, uninstall the addon and reinstall it directly. And the addon starts. Do not know why.

**Comment by matt8707** [comment]
> Add your repo and install the addon - Delete the hassio/addon/git folder from the addon, uninstall the addon and reinstall it directly. And the addon starts. Do not know why.

Can anyone else replicate this?

**Comment by churchofnoise** [comment]
I've been looking for the folder that I should delete using the file browser add on but haven't found it yet (any pointers @Gerrett84 ?), willing to try and replicate once that hurdle is taken!
(FYI, I'm running ha supervised on a Debian based system)

**Comment by schmierlappe** [comment]
I can confirm that after delete the git folder and reinstall ha-fusion the addon starts but then I run into this issues:

No existing file found for ./data/configuration.yaml
No existing file found for ./data/dashboard.yaml

when I remove everything

 - uninstall
 - remove repo
 - add repo
 -  install ha-fusion

then I have the EACCESS Error again

**Comment by matt8707** [comment]
Right, but I need **DETAILED** steps. Which git folder? Assume I'm an idiot! 😊

This is an example of a detailed answer

```bash
# login to server
ssh root@192.168.1.65 -p22

# addon container shell
docker exec -it addon_65f67727_ha_fusion

# remove git folder
rm -rf .git
```

**Comment by Gerrett84** [comment]
@matt8707 OK, tomorrow I'll write it in detail ☺️

@schmierlappe 
If the git folder was not created again during installation, simply restore the repo here using git clone and rename the folder accordingly
![Screenshot_20240103-232013.png](https://github.com/matt8707/addon-ha-fusion/assets/1213908/87b40eaa-a6f5-4644-9547-69f23a0e5091)



The data should appear after you open the UI for the first time Otherwise copy from the ha-fusion repo to ../addon/data yourself.
More details tomorrow 😉

![Screenshot_20240103-232156.png](https://github.com/matt8707/addon-ha-fusion/assets/1213908/38c5d937-14e3-4d67-bb07-0a18196fed0f)

**Comment by Gerrett84** [comment]
1. Add Add-on the Repo to Hassio
https://github.com/matt8707/addon-ha-fusion
![Bildschirmfoto vom 2024-01-04 15-06-24](https://github.com/matt8707/addon-ha-fusion/assets/1213908/457906f4-1582-462f-96c5-523d46f6c657)
2. Install the Add-on
3. Open a Terminal on host or Connect with SSH

`
#sudo needed
sudo bash

# cd to addon/git
cd /usr/share/hassio/addon/git

# remove addon-ha-fusion git folder
rm -rf 65f67727`

4. Got to Hassio and uninstall the Addon.
![Bildschirmfoto vom 2024-01-04 15-05-04](https://github.com/matt8707/addon-ha-fusion/assets/1213908/78cd6ebc-bee8-49b7-9942-47e21336162b)

5. Install the Addon again
6. Start the Addon - should work now.
7. Goto Terminal
`# bring back the git folder
git clone https://github.com/matt8707/addon-ha-fusion.git

#rename the folder
mv addon-ha-fusion 65f67727`

If you get the following error:
No existing file found for ./data/configuration.yaml
No existing file found for ./data/dashboard.yaml

open Fusion and set it up. If this doesn't work you can copy these files from here : https://github.com/matt8707/ha-fusion/tree/main/data to /usr/share/hassio/addon/data/656f7727_ha_fusion

**Comment by matt8707** [comment]
fixed [2024.1.1](https://github.com/matt8707/ha-fusion/releases/tag/2024.1.1) https://github.com/matt8707/addon-ha-fusion/commit/15f7279b88ec9df72f60cbe084538ff06ff3bb36
